add some coding standards convention and the developer's advice (#4)
NoemieDeraedt/SQA-Assignment-Group-10: Coding-standards.md

The right indentation is the one recognized by the good practices of the associated language. If there is none, the indentation associated with the project must be used to avoid compatibility problems between two developers.
![alt text](./Assets/https___bucketeer-e05bbc84-baa3-437e-9518-adb32be77984.s3.amazonaws.com_public_images_fb668c40-c3d1-4371-8eb2-cce6e179ebd2_800x437.jpeg)

### Improve the readability of the code

> Plusieurs technique sont possiblies afin de rendre le code plus lisible. Cela est très important, pour soit même lorsqu'on retourne sur notre code, ou alors pour un autre developper qui essaye de comprendre notre code rapidement. Voila quelque technique qui permette améliorer la lisibilité :

Reviewer: Text is in french ?

- Use short line lenght. It is easier to have logic and more comfortable for the eyes when the line is short. Usually the limit is 80 characters per line.
- Limit the number of structure nesting in a single function to 2.
![alt text](./Assets/coding-standards-nesting.jpeg)
- Do not use "if" when there are more 4 conditions and replace them with "switch cases" if it possible.

### Separation of the code by functions

> - It is important that a function makes a single spot and no more. This allows it to be reusable in several places and easier to read. For that you can limit the number of lines per function in order to break up the code.

Reviewer: ```suggestion
- It is important that a function is made only once. This allows it to be reusable in several places and easier to read. For that you can limit the number of lines per function in order to break up the code.
```

> - Refactoring of code to enhance the readalibilty without modifying the basic functionality

Reviewer: ```suggestion
- Refactoring of code to enhance the readability without modifying the basic functionality
```

> - Named function by a verb to describe the task

Reviewer: ```suggestion
- Naming function with a verb to describe the task
```

- What conventions are used or implemented within the organization
- What type of programs are being developed

> ### The develloper's advice:

Reviewer: ```suggestion
### The developer's advice:
```

> The coding standars is something that can make you lose time when writing but save a lot more afterwards. So it is very important to have a detailed coding style and to pay attention to it. It is also important to check the coding style during the code reviews in order to give good advices if needed.

Reviewer: ```suggestion
Coding standards are something that can make you lose time when writing but save a lot more afterwards. So it is very important to have a detailed coding style and to pay attention to it. It is also important to check the coding style during the code reviews in order to give good advices if needed.
```
